app/src/main/java/com/laborofloathing/diceroller/MainActivity.kt

```diff
@@ -7,17 +7,31 @@ import android.widget.Button
 import android.widget.ImageView
 import android.widget.Toast
 import androidx.core.view.isVisible
+import timber.log.Timber
 import java.util.*
 
 class MainActivity : AppCompatActivity() {
 
     private var numSides = 6
+    private var sideShowing = 1
     private lateinit var diceImage: ImageView
     private lateinit var decreaseButton: Button
+    private val NUM_SIDES_KEY = "NumOfSides"
+    private val SIDE_SHOWING = "SideShowing"
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+        Timber.i("onCreate called")
+
+        diceImage = findViewById<ImageView>(R.id.dice_image)
+
+        if (savedInstanceState != null) {
+            numSides = savedInstanceState.getInt(NUM_SIDES_KEY)
+            setDiceImage(savedInstanceState.getInt(SIDE_SHOWING))
```

Comment:
The reason you needed to initialize `diceImage` sooner after adding this is because this references `diceImage`.